components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/CodeTokenResponseValidator.java

```diff
@@ -58,6 +64,10 @@ private static boolean isContainOIDCScope(String scope) {
     @Override
     public void validateRequiredParameters(HttpServletRequest request) throws OAuthProblemException {
 
+        if (!StringUtils.isBlank(request.getParameter(REQUEST_URI))) {
+            requiredParams = new ArrayList<>(Arrays.asList(CLIENT_ID, RESPONSE_TYPE, SCOPE, REQUEST_URI));
+            notAllowedParams.add(REQUEST);
+        }
```

Comment:
Can you explain the reasoning behind this. It looks like we are marking the required params based on external input. IMO we shouldn't do validations based on external input. these make spaces for loopholes.